### PR TITLE
Exploring new settings design

### PR DESCRIPTION
**Pull request by FerrerFerreAlex**
Exploring Settings design. 

The idea is to experiment with the settings needed for constructing a class in Swan. 
In this case, we will start with DesignVariable Initializer.

### CODE REVIEW

File: Topology Optimization/Material/Material_Interpolation.m [hunk 1]
```diff
@@ -9,21 +9,21 @@
     end
     
     methods (Static)
-        function obj=create(ConstitutiveProperties, TypeOfMaterial, interpolation, pdim)
-            switch TypeOfMaterial
+        function obj=create(settings)
```
Comment (marcnunezc) [inline Topology Optimization/Material/Material_Interpolation.m]: How do we want to call the "settings" that enter in each class or function?

"settings"?
"input"?
"input_settings"?
"data_base"?

Comment (FerrerFerreAlex) [reply Topology Optimization/Material/Material_Interpolation.m]: I guess the proper long name for constructing a class should be "constructorParameters", but it is too long. Maybe
"constructParams"?
"cParams"?
"cP"?

Comment (Trujillo94) [reply Topology Optimization/Material/Material_Interpolation.m]: Config?

Comment (FerrerFerreAlex) [reply Topology Optimization/Material/Material_Interpolation.m]: constructorArguments?
cArg?
params?
conParams?
cSettings?

I guess cParams could be not bad, can be interpreted as configParams, constructorParams, createParams. Input is probably a bit redundant

**Comment by marcnunezc** [comment]
Hi!

I would say that right now there's a complete implementation of the new settings.

Apart from further clean up and more tests, my main concern right now is:

- How to deal with parameters that are common for several classes (think of coord or pdim). Which class should know and manage this kind of dependencies? Maybe using singleton classes?

File: ExploringSettings/Settings_TestExploringSettings.m [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    %
+    % Specific implementation of NumHomog settings for a specific case
+    % Here we set the specific parameters that configure our test
+    % This kind of files would replace our current Benchmark files
+    
+    properties
+    end
+    methods
+        function obj = Settings_TestExploringSettings()
+            obj.outFileName = 'test2d_micro';
```
Comment (marcnunezc) [inline ExploringSettings/Settings_TestExploringSettings.m]: This file would be the actual implementation of the parameters for an specific benchmark or test. These could also be read from a file (as discussed in our previous meeting).

**Pull request merged**